src/stemmabench/algorithms/stemma.py

```diff
@@ -0,0 +1,209 @@
+        """
+        raise NotImplementedError()
+
+    def dict(self, include_edges: bool = False) -> Dict[str, dict]:
+        """Return a dict representation of the tree.
+        Dict is empty until tree is fitted (fitting can be done using .fit() method)
+
+        ### Args:
+            - include_edges (bool): Indicates if the dictionary should include
+
+        ### Returns:
+            - dict: A dictionay representaion of the calling stemma tree.
+
+        ### Raises:
+            - RuntimeError: If the stemma has not been fited yet.
+        """
+        if not self.fitted:
+            raise RuntimeError("Stemma not fitted yet.")
+        return self.root.dict(include_edges)
+    
+    def __eq__(self, value: object) -> bool:
+        """Returns true if both stemmas are the same. !!! Only checks that both stemmas contain all the same texts !!!
+        
+        ### Args:
+            - value (object): The object to be compared to the calling Stemma.
+
+        ### Returns:
+            - bool: Value indicating if the calling object and the value parameter are equal.
+        """
+        if isinstance(value, Stemma):
+            # Check that both text_lookup exist and have same length.
+            if self.text_lookup and value.text_lookup and len(self.text_lookup) == len(value.text_lookup):
+                for lab in self.text_lookup:
+                    if value.text_lookup.get(lab) == None or not self.text_lookup[lab].__eq__(value.text_lookup[lab]):
+                        return False
+                return True
+        return False
+    
+    def __repr__(self) -> str:
+        """String representation of the tree."""
+        if self.fitted:
+            return "Tree(" + json.dumps(self.dict(), indent=2) + ")"
+        return "Empty"
+
+    def dump(self, folder: Union[str, None] = None) -> None:
+        """Dump the generated stemma into a folder:
+            - The texts in .txt file named after the manuscript label.
+            - The corresponding tree structure in edge file.
+            If folder is not specified will use the stemas path_to_folder attribute as path.
+
+        ### Args:
+            - folder (str, Optional): The folder where the text should be written.
+
+        ### Raises:
+            - ValueError: If folder is not specified and the folder_path variable is not set.
+        """
+        if not self.folder_path and not folder:
+            raise ValueError("This stemmas path_to_folder varaibles is not set. Therefor folder must be specified.")
+        if not folder and self.folder_path:
+            folder = self.folder_path
+        if not os.path.isdir(folder):
+                Path(folder).mkdir(exist_ok=True)
+        for key in self.text_lookup:
+            self.text_lookup[key].dump(folder)
+
+    def compute(self,
+                algo = None,
+                edge_file: Union[str, None] = None,
+                folder_path: Union[str, None] = None,
```

Comment:
Remove the folder_path argument, as this adds to much confusion regarding what is the actual parameter taken into account in the end.

Reply:
folder_path parameter removed